core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -319,10 +319,14 @@ private[spark] class Executor(
 
         case _: TaskKilledException | _: InterruptedException if task.killed =>
           logInfo(s"Executor killed $taskName (TID $taskId)")
+          // Reset the interrupted status of the thread to update the status
+          Thread.interrupted()
           execBackend.statusUpdate(taskId, TaskState.KILLED, ser.serialize(TaskKilled))
```

Comment:
Thanks @zsxwing for looking into the patch.

> What will happen if the thread is interrupted when execBackend.statusUpdate is running? I think the executor will still crash.


I do think this is a problem. I have handled it in the latest, Can you look into the changes?